<center><b><font size=5>Проект 2: Исследование надежности заемщиков</font></b></center>

<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li></ul></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Категоризация данных</a></span><ul class="toc-item"><li><span><a href="#Категория-доходов" data-toc-modified-id="Категория-доходов-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Категория доходов</a></span></li><li><span><a href="#Категория-покупок" data-toc-modified-id="Категория-покупок-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Категория покупок</a></span></li></ul></li><li><span><a href="#Исследовательский-анализ-данных" data-toc-modified-id="Исследовательский-анализ-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследовательский анализ данных</a></span><ul class="toc-item"><li><span><a href="#Зависимость-между-количеством-детей-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-количеством-детей-и-возвратом-кредита-в-срок-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Зависимость между количеством детей и возвратом кредита в срок</a></span></li><li><span><a href="#Зависимость-между-семейным-положением-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-семейным-положением-и-возвратом-кредита-в-срок-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Зависимость между семейным положением и возвратом кредита в срок</a></span></li><li><span><a href="#Зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок" data-toc-modified-id="Зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Зависимость между уровнем дохода и возвратом кредита в срок</a></span></li><li><span><a href="#Влияние-целей-кредита-на-его-возврат-в-срок" data-toc-modified-id="Влияние-целей-кредита-на-его-возврат-в-срок-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Влияние целей кредита на его возврат в срок</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

### Предобработка данных

In [103]:
# импортируем библиотеку
import pandas as pd

# загрузим данные 
data = pd.read_csv('data.csv')

In [104]:
# выведем первые 20 строчек датафрейма
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [105]:
# выведем основную информацию о датафрейме
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


#### Обработка аномальных значений

In [106]:
# выведем описание таблицы
data.describe()

children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  1.935100e+04  
mean           0.972544      0.080883  1.674223e+05  
std            1.420324      0.272661  1.029716e+05  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.030532e+05  
50%            0.000000      0.000000  1.450179e+05  
75%            1.000000      0.000000  2.034351e+05  
max            4.000000      1.000000  2.265604e+06

К аномалиям можно отнести отрицательное и слишком большое (20) количество детей, отрицательное количество дней трудового стажа в столбце `days_employed`. 

Заменим отрицательные значения в столбце `days_employed` на положительные.

In [107]:
data['days_employed'] = data['days_employed'].abs()

Для каждого типа занятости выведем медианное значение трудового стажа в днях.

In [108]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть. Тем более этот столбец не понадобится для исследования.

Проверим какое количество данных занимают строки с аномальным количеством детей.

In [109]:
print('Процент записей с аномальным количеством детей: {}'.format(round(data.query('children == -1 or children == 20').shape[0] / data.shape[0] * 100, 2)))

Процент записей с аномальным количеством детей: 0.57


Записи с аномальным количеством детей составляют лишь пол процента. Такие данные можно удалить.

In [110]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

In [111]:
# проверим изменения
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

Аномалии удалены

#### Удаление пропусков

In [112]:
# выведем количество пропущенных значений для каждого столбца
data.isna().sum()

children               0
days_employed       2162
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2162
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения — `days_employed` и `total_income`. Второй столбец хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [113]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.

In [114]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

In [115]:
# убедимся, что все пропуски заполнены
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

#### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income` на целочисленный.

In [116]:
data['total_income'] = data['total_income'].astype(int)

#### Обработка дубликатов

Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру

In [117]:
data['education'] = data['education'].str.lower()

In [118]:
# выведем количество явных дубликатов
data.duplicated().sum()

71

In [124]:
print('Процент явных дубликатов: {}'.format(round(data.duplicated().sum() / data.shape[0] * 100, 5)))

Процент явных дубликатов: 0.33174


Явные дубликаты составляют лишь 0.3% от общего количества записей. Их можно удалить.

In [125]:
data = data.drop_duplicates()

### Категоризация данных

#### Категория доходов

Создадим в датафрейме `data` столбец `total_income_category` с категориями:

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

In [17]:
# напишем функцию определения категорий
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [126]:
# применим функцию к новому столбцу
data['total_income_category'] = data['total_income'].apply(categorize_income)

#### Категория покупок

Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose'.

In [127]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

In [128]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [129]:
# применим функцию к новому столбцу
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Исследовательский анализ данных

#### Зависимость между количеством детей и возвратом кредита в срок

In [130]:
# просмотрим уникальные значения в столбце 'children'
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

In [131]:
# проверим, нет ли в столбце 'debt' лишних данных
data['debt'].unique()

array([0, 1], dtype=int64)

In [132]:
# сгруппируем данные из таблицы data по количеству детей с нахождением среднего значения невыплаты кредита в срок
data_child_debt = data.groupby('children')['debt'].agg(['sum', 'count', 'mean'])

In [133]:
# выведем таблицу на экран
data_child_debt

sum  count      mean
children                       
0         1063  14091  0.075438
1          444   4808  0.092346
2          194   2052  0.094542
3           27    330  0.081818
4            4     41  0.097561
5            0      9  0.000000

In [26]:
# переведем данные в новой таблице в проценты для более удобного восприятия информации
data_child_debt['mean'] = data_child_debt['mean'].apply(lambda x: format(x, '.2%'))

In [27]:
data_child_debt['mean'].sort_values(ascending=False)

children
4    9.76%
2    9.45%
1    9.23%
3    8.18%
0    7.54%
5    0.00%
Name: mean, dtype: object

**Вывод:** Проанализировав зависимость между количеством детей и возвратом кредита в срок, можно сделать вывод, что вероятность вернуть кредит у человека без детей выше, чем у тех, кто имеет 1-4 детей. Информации о клиентах с 5 детьми слишком мало (9 из более чем 20 тысячи записей), следовательно не стоит делать о таких клиентах каких-либо поспешных выводов. Те, у кого 3 ребенка, вовзращает кредит в срок чаще, чем те, у кого 1, 2 и 4 ребенка. Зависимость между количеством детей и возвратом кредита в срок присутствует

#### Зависимость между семейным положением и возвратом кредита в срок

In [134]:
# просмотрим уникальные значения в столбце 'family_status' 
data['family_status'].unique() 

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

In [135]:
# сгруппируем данные из таблицы data по семейному положению с нахождением суммы и количества значений невыплаты кредита в срок
data_family_status = data.groupby('family_status')['debt'].agg(['sum', 'count']) 

In [136]:
data_family_status

sum  count
family_status                    
Не женат / не замужем  273   2796
в разводе               84   1189
вдовец / вдова          63    951
гражданский брак       385   4134
женат / замужем        927  12261

In [137]:
# найдем вероятность невозврата кредита в срок в отдельном столбце 'percent', поделив значения столбца 'sum' на 'count'
data_family_status['percent'] = data_family_status['sum']/data_family_status['count'] 

In [138]:
# переведем данные в новом столбце в проценты для более удобного восприятия информации
data_family_status['percent'] = data_family_status['percent'].apply(lambda x: format(x, '.2%')) 

In [139]:
# отсортируем данные в таблице по вероятности невозврата кредита в срок по убыванию
data_family_status.sort_values(by='percent', ascending=False) 

sum  count percent
family_status                            
Не женат / не замужем  273   2796   9.76%
гражданский брак       385   4134   9.31%
женат / замужем        927  12261   7.56%
в разводе               84   1189   7.06%
вдовец / вдова          63    951   6.62%

**Вывод:** Кредитному отделу следует с осторожностью относиться к клиентам, которые не состоят или не состояли в браке. Согласно данным такие клиенты менее ответственные. Получается, что узаконенные отношения (включая и бывшие отношения) ведут к более ответственному отношению к своим обязательствам.

#### Зависимость между уровнем дохода и возвратом кредита в срок

In [140]:
# создадим сводную таблицу с индексами 'total_income_category' и значениями 'debt', применив функции 'sum', 'count', 'mean'  
data_pivot = data.pivot_table(index='total_income_category', values='debt', aggfunc=['sum', 'count', 'mean']) 

In [141]:
data_pivot

sum  count      mean
                       debt   debt      debt
total_income_category                       
A                         2     25  0.080000
B                       354   5014  0.070602
C                      1353  15921  0.084982
D                        21    349  0.060172
E                         2     22  0.090909

In [147]:
# соединим названия столбцов в одну ячейку
data_pivot.columns = [' '.join(col).strip() for col in data_pivot.columns.values] 

In [148]:
data_pivot

sum debt  count debt  mean debt
total_income_category                                 
A                             2          25   0.080000
B                           354        5014   0.070602
C                          1353       15921   0.084982
D                            21         349   0.060172
E                             2          22   0.090909

In [149]:
# переведем данные в новом столбце в проценты для более удобного восприятия информации
data_pivot['mean debt'] = data_pivot['mean debt'].apply(lambda x: format(x, '.2%')) 

In [150]:
# отсортируем данные в таблице по вероятности невозврата кредита в срок по убыванию
data_pivot.sort_values(by='mean debt', ascending=False) 

sum debt  count debt mean debt
total_income_category                                
E                             2          22     9.09%
C                          1353       15921     8.50%
A                             2          25     8.00%
B                           354        5014     7.06%
D                            21         349     6.02%

Как видно из таблицы, данное деление на категории слишком сильно расщепляет кредиторов. Категории "A" и "E" имеют 22 и 25 значений в таблице соответственно. Проведем новую категоризацию данных, уменьшив количество категорий.

In [151]:
# создадим функцию, создующую 3 категории дохода: до 50000(низкий), от 50001 до 100000(высокий) и от 100001(высокий)
def new_category(income): 
    if income <= 50000:
        return 'низкий доход'
    if 50001 <= income <= 100000:
        return 'средний доход'
    else:
        return 'высокий доход'

In [152]:
# созадим новый столбец 'new_income_category' и применим новую функцию категоризации дохода
data['new_income_category'] = data['total_income'].apply(new_category) 

In [153]:
# создадим сводную таблицу с индексами 'new_income_category' и значениями 'debt', применив функции 'sum', 'count', 'mean'  
data_pivot = data.pivot_table(index='new_income_category', values='debt', aggfunc=['sum', 'count', 'mean']) 

In [154]:
# соединим названия столбцов в одну ячейку
data_pivot.columns = [' '.join(col).strip() for col in data_pivot.columns.values] 

In [155]:
# переведем данные в столбце 'mean debt' в проценты для более удобного восприятия информации
data_pivot['mean debt'] = data_pivot['mean debt'].apply(lambda x: format(x, '.2%')) 

In [156]:
# отсортируем данные в таблице по вероятности невозврата кредита в срок по убыванию
data_pivot.sort_values(by='mean debt', ascending=False) 

sum debt  count debt mean debt
new_income_category                                
высокий доход            1378       16887     8.16%
средний доход             331        4073     8.13%
низкий доход               23         371     6.20%

**Вывод:** У людей с низким доходом наименьшая вероятность невозврата кредита в срок. Связано это с нежеланием портить кредитную историю для одобрения дальнейших кредитов. Но, как видно из таблицы, кредиторов с низким доходом в разы меньше, чем со средним и высоким доходом, чьи вероятности просроченности кредитов практически равны.

#### Влияние целей кредита на его возврат в срок

In [157]:
# создадим сводную таблицу с индексами 'purpose_category' и значениями 'debt', применив функции 'sum', 'count', 'mean'  
data_pivot = data.pivot_table(index='purpose_category', values='debt', aggfunc=['sum', 'count', 'mean']) 

In [158]:
# соединим названия столбцов в одну ячейку
data_pivot.columns = [' '.join(col).strip() for col in data_pivot.columns.values] 

In [159]:
# переведем данные в столбце 'mean debt' в проценты для более удобного восприятия информации
data_pivot['mean debt'] = data_pivot['mean debt'].apply(lambda x: format(x, '.2%')) 

In [160]:
# отсортируем данные в таблице по вероятности невозврата кредита в срок по убыванию
data_pivot.sort_values(by='mean debt', ascending=False) 

sum debt  count debt mean debt
purpose_category                                        
операции с автомобилем         400        4279     9.35%
получение образования          369        3988     9.25%
проведение свадьбы             183        2313     7.91%
операции с недвижимостью       780       10751     7.26%

**Вывод:** "Операции с недвижимостью" является самой маловероятной для просроченности кредита целью, с учетом того, что количество кредитов для данной цели в разы больше других целей кредитования. Чаще всего невозврат кредитов срок происходит в категориях "операции с автомобилем" и "получением образования".

### Общий вывод

    Было проведено исследование, чтобы выявить влияние разных факторов на погашение кредита в срок. Входные данные - статистика о платежеспособности клиентов. Результаты исследования будут учтены при построеннии модели кредитного скоринга - специальной системы, которая оценивает способность потенциального заемщика вернуть кредит банку.
    Перед проведением исследования были поставлены следующие гипотезы:
        - Гипотеза 1: "Между количеством детей и возвратом кредита в срок есть зависимость"
        - Гипотеза 2: "Между семейным положением и возвратом кредита в срок есть зависимость"
        - Гипотеза 3: "Между уровнем дохода и возвратом кредита в срок есть зависимость"
        - Гипотеза 4: "Между целью кредита и возвратом кредита в срок есть зависимость"
    В результате удалось выявить зависимости.
    Гипотеза 1 подтверждена: *для более точных выводов необходимо больше данных по клиентам с 5 детьми.
        •7.54% - доля просроченных кредитов у заемщиков без детей
        •8.18% - доля просроченных кредитов у заемщиков с 3 детьми
        •0% - доля просроченных кредитов у заемщиков с 5 детьми
        •9.23-9.76% - доля просроченных кредитов у заемщиков с 1, 2 и 4 детьми
    Рекомендация: провести дополнительное исследование на большей выборке данных.
    Гипотеза 2 подтверждена:
        •у не женатых клиентов(9.76%) и тех, у кого брак официально не зарегистрирован(9.31%), выше вероятность стать должниками
        •те, кто был женат(7.06%), овдовел(6.62%) или женат в данный момент(7.56%), имеют более низкую вероятность стать должниками
    Рекомендация: Между максимальным и минимальным процентов вероятности возврата кредита в срок всего 3%. Следовательно не стоит делать глобальных выводов из проверки данной гипотезы.
    Гипотеза 3 частично подтверждена
        •8.16% - доля просроченных кредитов у клиентов с высоким уровнем дохода
        •8.13% - доля просроченных кредитов у клиентов со средним уровнем дохода
        •6.20% - доля просроченных кредитов у клиентов с низким уровнем дохода
    Рекомендация: провести дополнительное исследование на большей выборке данных (особенно про людей с низким и средним уровнем дохода)
    Гипотеза 4 подтверждена
        •7.26% - доля просроченных кредитов, целью которых были операции с недвижимостью
        •7.91% - доля просроченных кредитов, целью которых были проведение свадьбы
        •9.25% - доля просроченных кредитов, целью которых были получение образования
        •9.35% - доля просроченных кредитов, целью которых были операции с автомобилем
        Сделать выводы об идеальном заемщике в данном случае сложно, так как в большинстве проверок гипотез слишком мало данных об определенной категории клиентов, чтобы делать выводы о кредитоспособности этих клиентов.